core/src/main/java/org/commonjava/maven/ext/core/impl/PreparseGroovyManipulator.java

```diff
@@ -0,0 +1,134 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.maven.ext.core.impl;
+
+import groovy.lang.GroovyShell;
+import groovy.lang.Script;
+import org.apache.maven.repository.DefaultMirrorSelector;
+import org.commonjava.maven.ext.common.ManipulationException;
+import org.commonjava.maven.ext.common.model.Project;
+import org.commonjava.maven.ext.core.ManipulationSession;
+import org.commonjava.maven.ext.core.groovy.InvocationPoint;
+import org.commonjava.maven.ext.core.groovy.InvocationStage;
+import org.commonjava.maven.ext.core.state.GroovyState;
+import org.commonjava.maven.ext.core.state.State;
+import org.commonjava.maven.ext.io.FileIO;
+import org.commonjava.maven.ext.io.ModelIO;
+import org.commonjava.maven.ext.io.PomIO;
+import org.commonjava.maven.ext.io.resolver.GalleyAPIWrapper;
+import org.commonjava.maven.ext.io.resolver.GalleyInfrastructure;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * {@link Manipulator} implementation that can resolve a remote groovy file and execute it on executionRoot. Configuration
+ * is stored in a {@link GroovyState} instance, which is in turn stored in the {@link ManipulationSession}.
+ */
+@Named("groovy-injection-preparse")
+@Singleton
+public class PreparseGroovyManipulator
+                extends BaseGroovyManipulator
+    //implements Manipulator
+{
+    @Inject
+    public PreparseGroovyManipulator( ModelIO modelIO, FileIO fileIO, PomIO pomIO )
+    {
+        super( modelIO, fileIO, pomIO );
+    }
+
+    /**
+     * Initialize the {@link GroovyState} state holder in the {@link ManipulationSession}. This state holder detects
+     * version-change configuration from the Maven user properties (-D properties from the CLI) and makes it available
+     * for later.
+     */
+    //@Override
+    public void init( final ManipulationSession session ) throws ManipulationException
+    {
+        final State state = new GroovyState( session.getUserProperties() );
+        this.session = session;
+        session.setState( state );
+
+        try
+        {
+            final GalleyInfrastructure infra = new GalleyInfrastructure( session, new DefaultMirrorSelector() );
+            infra.init( null, null, Files.createTempDirectory( "pme-" ).resolve( "cache-dir" ).toFile() );
+            final GalleyAPIWrapper wrapper = new GalleyAPIWrapper( infra );
+            modelIO = new ModelIO( wrapper );
+            fileIO = new FileIO( infra );
+        }
+        catch ( IOException e )
+        {
+            throw new ManipulationException( e.getMessage(), e );
+        }
```

Comment:
I think both of these above catch clauses could be removed if just `applyGroovyScripts` is used?